### DISCUSSION TITLE
Why are my reminders not working?

### DISCUSSION THREAD

**Discussion by hamhamhammchamm** [iSoron/uhabits#691] [Help & FAQ]
I'm on a Huawei phone and the push notifications for reminders doesn't seem to work unless I have the app open. Is there a way to fix this? Thank you.

**Comment by iSoron** [comment]
Hi @hamhamhammchamm, in our FAQ we list a few reasons why reminders may not work. See below. If none of these apply, please let us know. We may have a bug in the app.

* Loop will only show you a reminder if the task has not been already completed that day. For example, if your reminder is scheduled to show up at 8am, but you already added a checkmark manually (on the main app) at 7am, then the reminder will not appear.  The same is true for non-daily habits. If you have a weekly habit, and you complete it on a certain Monday, the reminder will only appear on the following Monday.
* When you create a habit, you can also choose what days are the reminder allowed to appear. If you create a weekly habit, but set the reminders to appear only on the weekends, for example, then they will never appear Monday to Friday, regardless of you having completed the habit during the weekend or not.
* If you are sure that the reminders should appear, but they don't, please, make sure that your phone has no task killers or RAM boosters installed. These apps are known to disable Loop's reminders. If your phone has an aggressive power saving mode, such as Samsung's Ultra Power Saving mode or Sony's Stamina Mode, please make sure that Loop is whitelisted.

**Reply by shaneknysh** [reply]
I recently moved to a TCL ultra 60 and notifications are basically dead now. 

I can't set the app to allow background process - the option isn't available in the OS. I set it via adb with no change.

I've set every habit to once a day but I never get notifications.

I can see some notifications in notification history but they never pop up.

I've reinstalled the app, cleared cache, backed up tasks, cleared tasks, recreated tasks, restored from backup...

**Comment by Jineshpanchal** [comment]
we are facing similar issue, some time reminder is not trigger and not visible on phone at all when screen is off or app is killed.

only when the phone screen is on, it is working fine.

we have tested this app in many diffrent phone and diffrent android version. Reminder is not working consistently. apart from remainder, everything is perfect.
